### ISSUE TITLE
Releasing SymPy 1.5

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#17877]
This issue is to track discussions about the SymPy 1.5 release. I will add more to this later.

**Comment by oscarbenjamin** [comment]
I've create a 1.5 branch. Any PRs now merged into master will *not* go in to SymPy 1.5. To get a PR merged for 1.5 it now needs to be based on the 1.5 branch. (I think it is possible to change this retrospectively in an existing PR.)

Aside: @moorepants and @smichr there are a number of stale branches in the main SymPy repo created by each of you. Those should be deleted unless they are still needed for an open PR.

**Comment by asmeurer** [comment]
Related: https://github.com/sympy/sympy/issues/17716

**Comment by oscarbenjamin** [comment]
I anticipate that a bunch more things will get merged before the release candidate but I wonder if it's worth doing a beta release right now. We could put up pre-release wheels on PyPI so that more people have time to test them before the main release.

**Comment by asmeurer** [comment]
It's up to you if you want to do it. In the past I haven't uploaded pre-releases to PyPI because pip used to install them by default, but I think that's been fixed now. If you give me your PyPI username I will add you.

**Comment by asmeurer** [comment]
I think at a minimum we should only do releases from the release script, even beta releases. So at least the release script checks will need to pass. The release script runs in a Docker container so that it is completely isolated and reproducible. Let me know if you have any issues running it or questions about the workflow (or suggestions on how to improve it).

**Comment by oscarbenjamin** [comment]
My PyPI username is oscarbenjamin. It might take me a little while to get everything set up as I don't think I can run docker on this machine.

**Comment by asmeurer** [comment]
OK. If you end up changing the Dockerfile you'll also need permissions for DockerHub so you can upload the newest build there. I can't think of any other services you'll need access for. I can take care of doing the announcements from the @sympy Twitter (I don't think it's easy to share a Twitter account).

**Comment by asmeurer** [comment]
If you think it's worth it, we can also look into making the release script run on some CI. There are a few CIs that support custom Docker images. I would try either GitHub Actions or CircleCI. That way you wouldn't need to run Docker yourself, which unfortunately does require sudo and also can take up a bit of disk space. 

Right now the final release artifacts are uploaded as a separate step for safety, since uploads are irreversible on PyPI. But that can potentially be done on CI as well. I know there are a few projects that do that. Or we can continue to do the final uploads manually, and just download the artifacts from the CI.

**Comment by oscarbenjamin** [comment]
I see these warnings from running the tests via the release script:
```
sympy/core/tests/test_args.py[901] /opt/conda/envs/sympy-tests/lib/python3.6/site-packages/theano/configdefaults.py:560: UserWarning: DeprecationWarning: there is no c++ compiler.This is deprecated and with Theano 0.11 a c++ compiler will be mandatory
  warnings.warn("DeprecationWarning: there is no c++ compiler."
WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python implementations. Performance will be severely degraded. To remove this warning, set Theano flags cxx to an empty string.
WARNING (theano.configdefaults): install mkl with `conda install mkl-service`: No module named 'mkl'
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:526: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:527: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:528: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:529: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:530: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:535: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
```
Some of those look like tensorflow using something deprecated in numpy. The first few look like Theano wants the test environment to have MKL and a g++ compiler.

**Comment by asmeurer** [comment]
See https://github.com/sympy/sympy/pull/14747#issuecomment-399619084. The tensorflow warnings might be new. Some of the warnings that are printed during the tests were unfortunately not possible for me to figure out how to remove them. If you can figure it out that's great. Theano in particular is quite annoying the way it works (it tries to compile something at *import time*). 

By the way, I forgot that the release script tests with all the optional dependencies installed. We probably need to update those with the newest optional dependencies from .travis.yml.

**Comment by asmeurer** [comment]
The theano compiler warnings should be fixable by installing stuff, either conda packages for the build or apt packages in the Dockerfile.

**Comment by oscarbenjamin** [comment]
Another warning:
```
sympy/printing/tests/test_theanocode.py[38] ................/opt/conda/envs/sympy-tests/lib/python3.6/site-packages/theano/gof/opt.py:251: UserWarning: WARNING: Supervisor is not added. Please build a FunctionGraphvia theano.compile.function_module.std_graph()or add the Supervisor class manually.
  sub_prof = optimizer.optimize(fgraph)
```

**Comment by oscarbenjamin** [comment]
```
sympy/solvers/tests/test_solvers.py[115] ....../root/sympy/sympy/solvers/solvers.py:998: UserWarning: 
Warning: Ignoring dict=True due to incompatible type of 0 < x.
  incompatible type of %s.''' % sol))
```

**Comment by oscarbenjamin** [comment]
```
==> WARNING: A newer version of conda exists. <==
  current version: 4.6.8
  latest version: 4.7.12

Please update conda by running

    $ conda update -n base conda
```

**Comment by asmeurer** [comment]
You can ignore the conda warning. If you really want it to go away you can rebuild the Docker container (that will also update all the apt packages).

**Comment by oscarbenjamin** [comment]
The release script now fails when building the docs:
```
PYTHONPATH=..: sphinx-build -b html -d _build/doctrees -W --keep-going src _build/html
Running Sphinx v2.2.1

Extension error:
Could not import extension sphinx_math_dollar (exception: No module named 'sphinx_math_dollar')
Makefile:49: recipe for target 'html' failed
make: *** [html] Error 2
activity-error:build_docs:activity failed with execption:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/rever/activity.xsh", line 83, in __call__
    self.func(*args, **kwargs)
  File "rever.xsh", line 104, in build_docs
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22304, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured="hiddenobject")
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22267, in run_subproc
    command.end()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15593, in end
    self._end(tee_output=tee_output)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15612, in _end
    self._raise_subproc_error()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15738, in _raise_subproc_error
    raise subprocess.CalledProcessError(rtn, spec.args, output=self.output)
subprocess.CalledProcessError: Command '['make', 'html']' returned non-zero exit status 2.
rewinding to afd5f1f4c9cfd36373cda7176881cb754e1194ab
```

**Comment by oscarbenjamin** [comment]
```diff
diff --git a/release/rever.xsh b/release/rever.xsh
index 6a643f44fb..a62e8f3a14 100644
--- a/release/rever.xsh
+++ b/release/rever.xsh
@@ -96,7 +96,7 @@ def wheel():
 
 @activity(deps={'_version'})
 def build_docs():
-    with run_in_conda_env(['sphinx', 'docutils', 'numpy', 'mpmath', 'matplotlib'],
+    with run_in_conda_env(['sphinx', 'docutils', 'numpy', 'mpmath', 'matplotlib', 'sphinx_math_dollar'],
         envname='sympy-release-docs'):
 
         cd doc
```

**Comment by oscarbenjamin** [comment]
I'm past the building the docs state of the release script. Now I've got here:
```
activity-start:compare_tar_against_git:starting activity compare_tar_against_git

Files in the tarball from git that should not be there:


Files in git but not in the tarball:

.ci/README.rst
.github/FUNDING.yml
MANIFEST.in
coveragerc_travis
sympy/plotting/tests/test_region_and.png
sympy/plotting/tests/test_region_not.png
sympy/plotting/tests/test_region_or.png
sympy/plotting/tests/test_region_xor.png

Files in the tarball but not in git:

doc/commit_hash.txt

Non-whitelisted files found or not found in the tarball
```

**Comment by oscarbenjamin** [comment]
Okay I'm at the end of the release script and I have SymPY 1.5 beta ready

**Comment by oscarbenjamin** [comment]
Some of the author names are clearly just Github usernames:
```
## Authors

The following people contributed at least one patch to this release (names are
given in alphabetical order by last name). A total of 135 people
contributed to this release. People with a * by their names contributed a
patch for the first time for this release; 71 people contributed
for the first time for this release.

Thanks to everyone who contributed to this release!

- Akash Nagaraj (akasnaga)*
- Shubham Abhang
- Abhi58*
- Sachin Agarwal
- Harsh Agarwal*
- Supreet Agrawal
- Abhinav Anand*
- Oscar Gerardo Lazo Arjona*
- arooshiverma*
- arun*
- Juan Barbosa*
- Oscar Benjamin
- Gabriel Bernardino*
- Parker Berry
- Bharat123rox*
- Johan Blåbäck
- Francesco Bonazzi
- Ondřej Čertík
- Arighna Chakrabarty
- Pengning Chao*
- Sean P. Cornelius*
- Corwinpro*
- ctsiagkalis*
- Björn Dahlgren
- Nabanita Dash
- Anway De*
- Gaurav Dhingra
- Mark Dickinson*
- dranknight09*
- erik*
- Zachariah Etienne
- eward
- Isuru Fernando
- Enric Florit*
- Mauro Garavello
- Matthias Geier
- Sourav Ghosh
- Lauren Glattly*
- Christoph Gohle*
- grassknoted*
- Oscar Gustafsson
- David Hagen*
- hannah-kari*
- Hou-Rui*
- Jack*
- Shubham Kumar Jha
- Ishan Joshi
- Aadit Kamat*
- kangzhiq
- Hannah Kari*
- Atharva Khare
- Cameron King*
- kmm555*
- Shivani Kohli
- George Korepanov*
- Amit Kumar
- Ritesh Kumar
- Samesh Lakhotia*
- Ronan Lamy
- S.Y. Lee
- Steven Lee*
- louisabraham*
- luzpaz
- Megan Ly*
- Morten Olsen Lysgaard*
- Nikhil Maan*
- Colin B. Macdonald
- Riccardo Magliocchetti*
- KIRTAN  MALI*
- Miguel Marco*
- Maxence Mayrand
- Pragyan Mehrotra
- Henry Metlov*
- Aaron Meurer
- Jogi Miglani
- Aaron Miller
- Sudhanshu Mishra
- Jason Moore
- Yeshwanth N*
- Akshay Nagar
- Sidhant Nagpal
- Prionti Nasir*
- Soniya Nayak*
- Abdullah Javed Nesar
- Kenneth Emeka Odoh*
- OrestisVaggelis*
- Ankit Raj Pandey
- pekochun*
- Ruslan Pisarev
- Tomasz Pytel*
- Samnan Rahee
- Shekhar Prasad Rajak
- Rohit Rango
- Joseph Redfern*
- Juha Remes
- Rimi*
- rimibis*
- riteshkumar*
- Faisal Riyaz
- Martin Roelfs
- Denys Rybalka*
- Denis Rykov*
- Samesh*
- Bendik Samseth*
- Gilles Schintgen*
- Stan Schymanski
- Daniel Sears*
- Stephan Seitz*
- Vighnesh Shenoy*
- Ritu Raj Singh
- Gagandeep Singh
- Jashanpreet Singh
- Sartaj Singh
- Rajiv Ranjan Singh
- Arun Singh
- Animesh Sinha
- Gleb Siroki
- Chris Smith
- Soniyanayak51*
- sqs*
- Ralf Stephan
- Kalevi Suominen
- Divyanshu Thakur
- TitanSnow*
- ivan tkachenko*
- Charalampos Tsiagkalis*
- Versus Void*
- Ethan Ward
- Jonathan Warner
- wate123*
- Srinivasa Arun Yeragudipati*
- yogesh1997*
- Velibor Zeli*
- Zhi-Qiang Zhou
- znxftw*
```
Is that something that should be fixed in mailmap?

Also what should be done with the SHA256 hashes:
```
ce87ab3ffe55ba1954fafdfa53980b247aa7390bb9bde169a4163a696873c368	sympy-1.5b1-py2.py3-none-any.whl
62fa5cd81d4afc62535657b1edf2132d08806c677a4e787c8f0f095d49d848ff	sympy-1.5b1.tar.gz
eeba3e7551b2746efb71587656af503b11add8a9819ae3ff1143b19c08e6306b	sympy-docs-html-1.5b1.zip
d940a996a06cb27c2ce51dc4ae778062041f204c1319ac5b283fde503cbb3b23	sympy-docs-pdf-1.5b1.pdf
```

**Comment by oscarbenjamin** [comment]
There are some comments in #17914 that I should follow up on before final release

**Comment by asmeurer** [comment]
That's whatever name comes from the commit metadata, or the GitHub metadata if they committed through the GitHub interface. You can ask people what name they prefer to have in the file. Some people want to use a pseudonym only and that's fine.

**Comment by asmeurer** [comment]
The .mailmap is how you fix that. 

The hashes are just there so you can verify the files are the correct ones. The script adds those to the release text on GitHub automatically when you run the upload command.

**Comment by oscarbenjamin** [comment]
I can't immediately figure out where dranknight09 comes from in the Author list above

**Comment by oscarbenjamin** [comment]
I've emailed all contributors whose full names we don't have to ask if they want their full name included in the release notes.

The only one I couldn't chase up at all was dranknight09

**Comment by oscarbenjamin** [comment]
I think I'm set to release 1.5 beta 1 tomorrow.

**Comment by moorepants** [comment]
This was introduced in the last GSoC: https://gitlab.com/sympy/autolev-test-examples

It has tests for SymPy that rely on proprietary code. We somehow need to run these tests before releases.

**Comment by asmeurer** [comment]
> I think I'm set to release 1.5 beta 1 tomorrow.

That's great!

> This was introduced in the last GSoC: https://gitlab.com/sympy/autolev-test-examples

> It has tests for SymPy that rely on proprietary code. We somehow need to run these tests before releases.

@oscarbenjamin if you give me your GitLab username I can add you.

If this repo needs to be tested, we need to figure out some way to do it automatically. Can we set up a recurring CI on the repo?

**Comment by moorepants** [comment]
> Can we set up a recurring CI on the repo?

Seems like we'd need the release script here to open a PR there that would trigger CI? Not sure if you can setup a CI run like a cron job.

It'd be nice if we could give travis access to the repo somehow and it pulls the code in the sympy tests here.

**Comment by asmeurer** [comment]
I mean to make it so that GitLab CI runs the tests every day or so. I think it supports that. The only question is how to report it if there is a failure. What should it do?

**Comment by oscarbenjamin** [comment]
> > This was introduced in the last GSoC: https://gitlab.com/sympy/autolev-test-examples
> 
> > It has tests for SymPy that rely on proprietary code. We somehow need to run these tests before releases.
> 
> @oscarbenjamin if you give me your GitLab username I can add you.

I've just signed up to gitlab with the username oscarbenjamin.

**Comment by oscarbenjamin** [comment]
Previous beta release have used tags like `sympy-0.6.3.beta1`. I can do that but the version I have put in `sympy/release.py` is `1.5b1` rather than `1.5.beta1 as specified in PEP 440: https://www.python.org/dev/peps/pep-0440/#pre-releases

Before I push the tag is it important that the version string in the git tag exactly matches the version in `sympy/release.py`?

**Comment by oscarbenjamin** [comment]
The `__version__` string used in the past has also been of that form e.g.:
```
>>> import sympy
>>> sympy.__version__
'0.6.3.beta1'
```

**Comment by oscarbenjamin** [comment]
It seems that SymPy started using PEP 440 version specifiers as of `1.1.1rc1`:
```
sympy-0.5.0           sympy-0.5.6           sympy-0.6.4           sympy-0.6.6           sympy-0.7.2.rc1       sympy-0.7.6.rc1       sympy-1.3 
sympy-0.5.10          sympy-0.5.7           sympy-0.6.4.beta1     sympy-0.6.6.rc1       sympy-0.7.3           sympy-0.7.6.rc2       sympy-1.3rc1 
sympy-0.5.11          sympy-0.5.8           sympy-0.6.4.beta2     sympy-0.6.7           sympy-0.7.3.rc1       sympy-1.0             sympy-1.4 
sympy-0.5.12          sympy-0.5.9           sympy-0.6.4.beta3     sympy-0.6.7-merge     sympy-0.7.4           sympy-1.0.rc1         sympy-1.4rc1 
sympy-0.5.13          sympy-0.6.0           sympy-0.6.5           sympy-0.7.0           sympy-0.7.4.1         sympy-1.1             sympy-1.4rc2 
sympy-0.5.14          sympy-0.6.1           sympy-0.6.5-beta1     sympy-0.7.0.rc1       sympy-0.7.4.rc1       sympy-1.1.1           sympy-release-conda 
sympy-0.5.15          sympy-0.6.2           sympy-0.6.5.beta2     sympy-0.7.0.rc2       sympy-0.7.5           sympy-1.1.1rc1        
sympy-0.5.3           sympy-0.6.3           sympy-0.6.5.beta3     sympy-0.7.1           sympy-0.7.5.rc1       sympy-1.1rc1          
sympy-0.5.4           sympy-0.6.3.beta1     sympy-0.6.5.rc1       sympy-0.7.1.rc1       sympy-0.7.6           sympy-1.2             
sympy-0.5.5           sympy-0.6.3.beta2     sympy-0.6.5.rc2       sympy-0.7.2           sympy-0.7.6.1         sympy-1.2rc1      
```
I'll go with that then and push the tag.

**Comment by asmeurer** [comment]
The tag and the version should match.

**Comment by oscarbenjamin** [comment]
I've pushed the beta release to PyPY with twine and announced it on the mailing list.

The release README refers to an upload command that isn't actually present in the rever script. I can see the corresponding command in the fabfile but don't have vagrant set up to be able to use the fabfile so I've manually used twine (it's just two commands). I guess that needs adding to the release script.

Having installed this back from pip I ran sympy.test and get a failure:
```
>>> sympy.test()
...
______________________________________________________________________________________________ xpassed tests _______________________________________________________________________________________________
sympy/core/tests/test_arit.py: test_failing_Mod_Pow_nested
sympy/integrals/tests/test_heurisch.py: test_pmint_erf
sympy/printing/tests/test_tree.py: test_print_tree_MatAdd
sympy/utilities/tests/test_wester.py: test_V12

____________________________________________________________________________________________________________________________________________________________________________________________________________
__________________________________________________________________________ sympy/utilities/tests/test_code_quality.py:test_files ___________________________________________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/test_venv/venv/lib/python3.7/site-packages/sympy/utilities/tests/test_code_quality.py", line 240, in test_files
    check_files(top_level_files, test)
  File "/Users/enojb/current/sympy/test_venv/venv/lib/python3.7/site-packages/sympy/utilities/tests/test_code_quality.py", line 125, in check_files
    file_check(fname)
  File "/Users/enojb/current/sympy/test_venv/venv/lib/python3.7/site-packages/sympy/utilities/tests/test_code_quality.py", line 146, in test
    test_this_file(fname, test_file)
  File "/Users/enojb/current/sympy/test_venv/venv/lib/python3.7/site-packages/sympy/utilities/tests/test_code_quality.py", line 165, in test_this_file
    assert False, message_test_suite_def % (fname, idx + 1)
AssertionError: Function should start with 'test_' or '_': /Users/enojb/current/sympy/test_venv/venv/lib/python3.7/site-packages/isympy.py, line 176

======================================= tests finished: 8481 passed, 1 failed, 334 skipped, 379 expected to fail, 4 expected to fail but passed, in 3987.68 seconds ========================================
DO *NOT* COMMIT!
False
```
Apparently the code quality test thinks that isympy.py in site-packages is a test file and is complaining about the `main` function there.

Also there are a few XFAILS that should be removed.

**Comment by moorepants** [comment]
That could work (if it is possible), but wouldn't it be better if Travis can run the tests here? I think if Travis can get access to clone the repo from gitlab, then we can test it here. It is fine if the files exist on the travis containers, as noone can get to them (at easily I guess). I don't really understand how these secret keys work (like doctr), would that be possible?

**Comment by asmeurer** [comment]
> The release README refers to an upload command that isn't actually present in the rever script. I can see the corresponding command in the fabfile but don't have vagrant set up to be able to use the fabfile so I've manually used twine (it's just two commands). I guess that needs adding to the release script.

I forgot that I never implemented that. Yeah, just use twine manually for now. The fabfile is the old release script. We should delete it to avoid confusion. 

> That could work (if it is possible), but wouldn't it be better if Travis can run the tests here? I think if Travis can get access to clone the repo from gitlab, then we can test it here. It is fine if the files exist on the travis containers, as noone can get to them (at easily I guess). I don't really understand how these secret keys work (like doctr), would that be possible?

Secret keys aren't available on pull request builds. So the tests would only run on master builds, which means people wouldn't really notice them failing (but maybe they would more than if the tests are running on GitLab CI). 

I don't know if GitHub actions has the same secret key restrictions. We could look into it.

**Comment by oscarbenjamin** [comment]
It doesn't look like we have much feedback on the beta so maybe it's time to do a release candidate.

**Comment by oscarbenjamin** [comment]
There have been a couple of minor PRs merged since the release candidate (#18000 and #17984) but I don't think we need another release candidate so I will make the final release of 1.5 shortly.

**Comment by oscarbenjamin** [comment]
I'm building the final release files now.

I've opened #18018 to merge changes from the 1.5 release branch back into master

**Comment by asmeurer** [comment]
I just noticed a tag was never pushed for the rc1.

**Comment by oscarbenjamin** [comment]
> I just noticed a tag was never pushed for the rc1.

I'll push that soon once I've got the release script working.

I hit a snag with the release script:
```
activity-start:test_tarball27:starting activity test_tarball27
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 21877, in _run_binary
    p = self.cls(self.cmd, bufsize=bufsize, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 14059, in __init__
    *args, stdin=stdin, stdout=stdout, stderr=stderr, **kwargs
  File "/opt/conda/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/opt/conda/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'create': 'create'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15332, in __init__
    proc = spec.run(pipeline_group=pipeline_group)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 21866, in run
    p = self._run_binary(kwargs)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 21888, in _run_binary
    raise XonshError(e)
xonsh.__amalgam__.XonshError: xonsh: subprocess mode: command not found: create
Did you mean the following?
    cweave:  Command (/usr/bin/cweave)
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 14887, in run
    r = self.f(self.args, sp_stdin, sp_stdout, sp_stderr, spec, spec.stack)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 14680, in proxy_two
    return f(args, stdin)
  File "/opt/conda/lib/python3.6/site-packages/xontrib/xonda.xsh", line 171, in _xonda
    @$(which -s conda) @(args)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22304, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured="hiddenobject")
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22267, in run_subproc
    command.end()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15593, in end
    self._end(tee_output=tee_output)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15612, in _end
    self._raise_subproc_error()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15738, in _raise_subproc_error
    raise subprocess.CalledProcessError(rtn, spec.args, output=self.output)
subprocess.CalledProcessError: Command '['create', '-y', '-n', 'test-install-2.7', 'python=2.7']' returned non-zero exit status 1.
activity-error:test_tarball27:activity failed with execption:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/rever/activity.xsh", line 83, in __call__
    self.func(*args, **kwargs)
  File "rever.xsh", line 129, in test_tarball27
  File "rever.xsh", line 289, in test_tarball
  File "/opt/conda/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/opt/conda/lib/python3.6/site-packages/rever/conda.xsh", line 31, in run_in_conda_env
    conda create -y -n @(envname) @(packages)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22304, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured="hiddenobject")
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 22267, in run_subproc
    command.end()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15593, in end
    self._end(tee_output=tee_output)
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15612, in _end
    self._raise_subproc_error()
  File "/opt/conda/lib/python3.6/site-packages/xonsh/__amalgam__.py", line 15738, in _raise_subproc_error
    raise subprocess.CalledProcessError(rtn, spec.args, output=self.output)
subprocess.CalledProcessError: Command '['conda', 'create', '-y', '-n', 'test-install-2.7', 'python=2.7']' returned non-zero exit status 1.
rewinding to 7f92c96417d066700d096105282d6fe47ea085ef
```
I'm not sure why that failed. I guess it must have worked when I ran the release script before and I don't think that anything has been changed

**Comment by asmeurer** [comment]
Looks like it might be a rever bug. @scopatz any thoughts?

**Comment by oscarbenjamin** [comment]
I just re-ran the release script and it completed with no errors...

**Comment by asmeurer** [comment]
I updated conda-forge https://github.com/conda-forge/sympy-feedstock/pull/19.

**Comment by oscarbenjamin** [comment]
> I updated conda-forge [conda-forge/sympy-feedstock#19](https://github.com/conda-forge/sympy-feedstock/pull/19).

That should be added to release/README (there are other changes to be made)

**Comment by asmeurer** [comment]
> That should be added to release/README (there are other changes to be made)

The steps for updating it are simple, because the bot updates automatically. So it's basically, wait until the bot notices the release on PyPI (usually takes less than an hour), then merge the PR. You have to be a maintainer on the feedstock (it would also be good to list in the release README the places you need access to).

**Comment by oscarbenjamin** [comment]
One thing I haven't figured out how to do is update SymPy live. II followed the instructions in the readme to update the submodule and increased the google apps number. Does gapps need to be told to deploy the changes somehow?

**Comment by oscarbenjamin** [comment]
I've merged #18063 into the 1.5 branch in preparation for a 1.5.1 release. I've opened #18067 to bump the version number in preparation for a bugfix release.

Should we wait to see if more bugs appear?

**Comment by oscarbenjamin** [comment]
I still haven't figured out how to update sympy live

**Comment by asmeurer** [comment]
You probably don't even have access to SymPy Live. I can look into doing it if you make a PR. Unfortunately the auto test deploy is only enabled for SymPy Gamma. 

SymPy Live is a major problem anyway, because it's still stuck on Python 2 and it doesn't seem easy to move to Python 3.

**Comment by asmeurer** [comment]
> Should we wait to see if more bugs appear?

I don't see any harm in waiting a little bit. Unless you'd prefer to make potentially multiple point releases if other issues come up.

**Comment by oscarbenjamin** [comment]
I have the release files prepared so that I could make a release now. Currently it has fixes for #18056 and #18079. I'm not aware of any other regressions or new issues reported yet.

**Comment by oscarbenjamin** [comment]
I think it's time to release 1.5.1. I'll do it tomorrow

**Comment by asmeurer** [comment]
Sounds good to me.

**Comment by cbm755** [comment]
It looks like https://github.com/sympy/sympy/releases is *not* up to date for 1.5.1.

What are the official tar.gz files and what are their SHA?   I see sympy-1.5.1.tar.gz on PyPI and Fedora has checked in something but their SHA do not match...

**Comment by cbm755** [comment]
For previous releases, the files on https://github.com/sympy/sympy/releases seem to match the ones on PyPI so I think I can take the PyPI 1.5.1.tar.gz as definitive.  Nonetheless we should fix the releases page...

**Comment by asmeurer** [comment]
The SHA1s for the release files were posted on the mailing list. I guess @oscarbenjamin didn't upload them to GitHub yet.

**Comment by oscarbenjamin** [comment]
Oh sorry I should do the Github release. I'll do that shortly

**Comment by asmeurer** [comment]
@oscarbenjamin did you use the release scripts to update the documentation? I noticed that the documentation URL for docs.sympy.org goes to https://docs.sympy.org/1.5.1/. But the latest docs should be at https://docs.sympy.org/latest, with https://docs.sympy.org/1.5.1/ as a redirect to it. The reason is that we want to minimize the number of links that exist in the wild that point to a specific version of the docs (I also think we should just remove old versions of the docs https://github.com/sympy/sympy_doc/pull/33).

**Comment by oscarbenjamin** [comment]
I ran the release script but it screwed up: I had a missing dependency so it aborted first time. After fixing that and rerunning the script the second run went over the top of the first and committed and pushed everything. I think that I should have recloned the docs before rerunning the script. Somewhere in the script it does something like `cp -r dir1 dir2` which should create a new directory `dir2` but actually puts everything inside `dir2` if `dir2` already exists.

In the end I tried to fix it up manually but I guess I didn't do it right.
